ports/raspberrypi/common-hal/rotaryio/IncrementalEncoder.c

```diff
@@ -60,19 +60,26 @@ STATIC void incrementalencoder_interrupt_handler(void *self_in);
 void common_hal_rotaryio_incrementalencoder_construct(rotaryio_incrementalencoder_obj_t *self,
     const mcu_pin_obj_t *pin_a, const mcu_pin_obj_t *pin_b) {
     mp_obj_t pins[] = {MP_OBJ_FROM_PTR(pin_a), MP_OBJ_FROM_PTR(pin_b)};
+    bool swap = false;
     if (!common_hal_rp2pio_pins_are_sequential(2, pins)) {
-        mp_raise_RuntimeError(translate("Pins must be sequential"));
+        pins[0] = MP_OBJ_FROM_PTR(pin_b);
+        pins[1] = MP_OBJ_FROM_PTR(pin_a);
+        swap = true;
+        if (!common_hal_rp2pio_pins_are_sequential(2, pins)) {
+            mp_raise_RuntimeError(translate("Pins must be sequential"));
+        }
     }
 
+    self->swap = swap;
```

Comment:
Why do we need to save this? I was thinking we'd just need to support both ordering but not change how position changes.

Reply:
it was part of the original request, as I understood it.  Otherwise, the user could just reverse the pins themselves in the constructor.

Reply:
Is the expectation the position will always be a difference to their counter? Is there a need to make direction reversible on the fly? Is there a need to see a roll over point internally? Is there a need to add scaling for revolutions? Can someone implement a new class based on this?

I know jepler's solution had better performance than the solution I was recommending. However there could be different use cases, which I had not considered before. I guess my question is what is the expectation for the user and API?

You could use the position variable as a counter and avoid using it in application logic. If you are porting to different hardware you could edit the constructor. Maybe the user of a GUI wants it one way so the GUI exposes that? However maybe you want to keep counting forwards even though the direction changed? 

If kept difference based you could retrofit all of this in application logic. However that logic could be ugly.

Reply:
I'd rather not add switching counting direction because you can do it outside of this class. My understanding what just that the pin order could be swapped.